modules/claims_api/app/controllers/claims_api/application_controller.rb

```diff
@@ -39,19 +39,26 @@ def source_name
 
     private
 
+    # rubocop:disable Metrics/CyclomaticComplexity
+    # rubocop:disable Metrics/PerceivedComplexity
     def find_claim
       claim = ClaimsApi::AutoEstablishedClaim.find_by(id: params[:id], source: source_name)
 
       if claim && claim.status == 'errored'
         fetch_errored(claim)
       elsif claim && claim.evss_id.nil?
         render json: claim, serializer: ClaimsApi::AutoEstablishedClaimSerializer
-      else
+      elsif /^\d{2,20}$/.match?(params[:id])
         evss_claim = claims_service.update_from_remote(claim.try(:evss_id) || params[:id])
         # Note: source doesn't seem to be accessible within a remote evss_claim
         render json: evss_claim, serializer: ClaimsApi::ClaimDetailSerializer
+      else
+        render json: { errors: [{ status: 404, detail: 'Claim not found' }] },
+               status: :not_found
       end
     end
+    # rubocop:enable Metrics/CyclomaticComplexity
+    # rubocop:enable Metrics/PerceivedComplexity
```

Comment:
i can take a look at this more if disabling this bothers yall. i spent some time trying to think of a clean way to break this out and anything that came to mind just felt more confusing than how it's laid out here.